raidz time-dependent geometry

ahrens opened the pull request:
extensions to #9:

* one entry per expansion (no initial entry)
* new layout starts in the *next* txg
* rename some things

Note: this doesn't save any space, we need to update vdev_raidz_asize() to change the amount of space allocated.

ahrens commented:
@fuporovvStack I rebased this

fuporovvStack commented on module/zfs/vdev_raidz.c:
```diff
@@ -3624,7 +3611,7 @@ vdev_raidz_get_tsd(spa_t *spa, nvlist_t *nv)
+	avl_create(&vdrz->vd_expand_txgs, vedv_raidz_reflow_compare,
 	    sizeof (reflow_node_t), offsetof(reflow_node_t, re_link));
 
 	error = nvlist_lookup_uint64_array(nv, ZPOOL_CONFIG_RAIDZ_EXPAND_TXGS,
```
Seems like, there is an issue in case of export/import during first expansion, when tree is empty.
Possible the next check could be added:
+       } else {
+               if (vdrz->vn_vre.vre_offset != UINT64_MAX)
+                       vdrz->vd_original_width-=1;
        }

ahrens replied:
Good point.  I think that problem would occur when importing during any expansion, not just the first.

ahrens commented:
I've addressed the remaining issues with this.  However, I did have to disable vdev_raidz_io_verify(), because vdev_xlate() doesn't really work for raidz's that are in the middle of being expanded (it doesn't take into account the per-row old vs new locations).
